lib/src/cache_manager.dart

```diff
@@ -88,33 +89,33 @@ class FirebaseImageCacheManager {
       where: 'uri = ?',
       whereArgs: [uri],
     );
-    if (maps.length > 0) {
-      FirebaseImageObject returnObject =
-          FirebaseImageObject.fromMap(maps.first);
+    if (maps.isNotEmpty) {
+      final returnObject = FirebaseImageObject.fromMap(maps.first);
       returnObject.reference = getImageRef(returnObject, image.firebaseApp);
-      if (CacheRefreshStrategy.BY_METADATA_DATE == this.cacheRefreshStrategy) {
-        checkForUpdate(returnObject, image); // Check for update in background
+      if (CacheRefreshStrategy.BY_METADATA_DATE == cacheRefreshStrategy) {
+        await checkForUpdate(
```

Comment:
This should be `unawaited(checkForUpdate( returnObject, image)); // Check for update in background` as otherwise it is blocking and not checking in the background.